criu/include/page-pipe.h

```diff
@@ -19,7 +20,7 @@ struct kernel_pipe_buffer {
  * fails very often, so we need to restrict the pipe capacity to not
  * allocate big chunks.
  */
-#define PIPE_MAX_SIZE ((1 << PAGE_ALLOC_COSTLY_ORDER) * PAGE_SIZE /	\
+#define PIPE_MAX_SIZE rounddown_pow_of_two((1 << PAGE_ALLOC_COSTLY_ORDER) * PAGE_SIZE /        \
 			sizeof(struct kernel_pipe_buffer))
```

Comment:
This means that `PIPE_MAX_SIZE` is calculated each time it's used, isn't it?
Maybe it's worth making PIPE_MAX_SIZE a variable and initialize it early.
Another thing, that we probably do not need all the highly optimized versions of `rounddown_pow_of_two()` and its dependencies. If we evaluate this once using a plain C implementation should suffice.

Reply:
> This means that `PIPE_MAX_SIZE` is calculated each time it's used, isn't it?

As far as I understand - it will not be calculated each time because in `rounddown_pow_of_two` implementation there is `__builtin_constant_p` check for the compile-time constant case.

>Another thing, that we probably do not need all the highly optimized versions of 

Good notice. I'm also thinking about that when realized how much lines of code I've copied from Linux kernel ;)